compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -1381,11 +1450,13 @@ private boolean hasAnnotation(List<BLangAnnotationAttachment> annotationAttachme
      * @param varNode       Variable node to be updated.
      * @param taintedStatus Tainted status.
      */
-    private void setTaintedStatus(BLangVariable varNode, boolean taintedStatus) {
-        if (!nonOverridingAnalysis) {
-            varNode.symbol.tainted = taintedStatus;
-        } else if (!varNode.symbol.tainted) {
-            varNode.symbol.tainted = taintedStatus;
+    private void setTaintedStatus(BLangVariable varNode, Boolean taintedStatus) {
+        if (taintedStatus != null) {
+            if (!nonOverridingAnalysis) {
+                varNode.symbol.tainted = taintedStatus;
+            } else if (!varNode.symbol.tainted) {
+                varNode.symbol.tainted = taintedStatus;
+            }
```

Comment:
Can we avoid this duplicated code block in two setTaintedStatus methods?

Reply:
This was changed with the enum change done and move to a method.